tfutils/base.py

```diff
@@ -97,83 +98,131 @@ def __init__(self,
+                                                     port=self.load_port)
+        else:
+            self.load_conn = self.conn
+        self.load_collfs = gridfs.GridFS(self.load_conn[self.load_dbname], 
+                                             self.load_collname)
+        load_recent_name = '_'.join([self.load_dbname,
+                                     self.load_collname,
+                                     self.load_exp_id,
+                                     '__RECENT'])
+        self.load_collfs_recent = gridfs.GridFS(self.load_conn[load_recent_name])
 
         if cache_dir is None:
             self.cache_dir = os.path.join(os.environ['HOME'],
                                           '.tfutils',
-                                          '%s:%d' % (host, port),
-                                          dbname,
-                                          collname,
-                                          exp_id)
+                                          '%s:%d' % (self.host, self.port),
+                                          self.dbname,
+                                          self.collname,
+                                          self.exp_id)
         else:
             self.cache_dir = cache_dir
         if not os.path.isdir(self.cache_dir):
             os.makedirs(self.cache_dir)
+        
+    def load_rec(self):
+        #first try and see if anything with the save data exists, since obviously
+        #we dont' want to keep loading from the original load location if some work has
+        #already been done
+        load = self.load_from_db({'exp_id': self.exp_id},
+                                 cache_model=True)
+        #if not, try loading from the loading location
+        if not load and not self.sameloc:
+            load = self.load_from_db(self.load_query,
+                                     cache_model=True,
+                                     collfs=self.load_collfs,
+                                     collfs_recent=self.load_collfs_recent)
+            if load is None:
+                raise Exeption('You specified load parameters but no record was found with the given spec.')
+        self.load_data = load
 
-        self.load_model(force_fetch=force_fetch)
-
-        self.start_time_step = time.time()  # start timer
-
-    def load_model(self, force_fetch=False):
+    def initialize(self):
         """
         Fetches record then uses tf's saver.restore
         """
         # fetch record from database and get the filename info from record
-        if self._restore:
-            load = self.load_from_db({'exp_id': self.exp_id,
-                                      'saved_filters': True},
-                                     cache_model=True,
-                                     force_fetch=force_fetch)
-            if load is not None:
-                rec, cache_filename = load
+        tf_saver = self.tf_saver
```

Comment:
Well, I need access to tf_saver below and I didn't want to keep typing "self.tf_saver" ... I guess since it's only used once that's stupid. so I can remove the line.